src/keyszer/lib/key_context.py

```diff
@@ -28,4 +28,15 @@ def x_error(self):
 
     @property
     def device_name(self):
+        self._device_name = self.device.name
         return self._device_name
+
+    @property
+    def capslock_on(self):
+        self._capslock_on = True if 1 in self.device.leds() else False
+        return self._capslock_on
```

Comment:
What is the value of the instance variable vs just returning the expression directly?

Reply:
Apparently it was redundant. I thought of return as being the end of a long process, but I guess in this case it doesn't need to be since I consolidated the if-else into the one-line "ternary" form.